website/src/auth.tsx

```diff
@@ -40,7 +42,7 @@ export const UserProvider = (props: { children: any }) => {
         }
       })
     }
-  }, [user])
+  }, [token])
```

Comment:
Why did this change exactly? Since the hook uses `user`, it becomes less easy to understand with this mismatch. If it's necessary, then add a comment in the code explaining why.

Reply:
I noticed while testing out the editable word functionality, that after some period of time while logged in, the user's token will become invalid. I made this `useEffect` originally as a solution to that problem, but realized the only state that changes is `token` and not the `user` object so the dependency hook needs `token` to work properly.